crates/fraud_proofs/global_merkle_root/storage/src/update.rs

```diff
@@ -528,6 +542,35 @@ mod tests {
         assert!(coin_doesnt_exist_after_process_input);
     }
 
+    #[test]
+    fn process_transaction__should_store_processed_transaction() {
```

Comment:
i think we need some unhappy path tests too for this

Reply:
Sure, added a test case in https://github.com/FuelLabs/fuel-core/pull/2650/commits/5dc7e60add2dded98c236b08480b993c3aeedea8 for when the transaction is duplicate which is the only failure mode currently - except for storage failures but that shouldn't be tested here imo.